src/kagglehub/exceptions.py

```diff
@@ -58,9 +58,11 @@ def kaggle_api_raise_for_status(response: requests.Response, resource_handle: Op
         response.raise_for_status()
     except requests.HTTPError as e:
         message = str(e)
-        server_error_message = response.json().get("message", "")
-        if server_error_message:
-            server_error_message = f"The server reported the following issues: {server_error_message}\n"
+        server_error_message = ""
+        if response.headers.get("Content-Type") == "application/json":
```

Comment:
Should probably do a case insensitive check. 

add a try/except block to catch the `requests.exception.JSONDecodeError`: https://requests.readthedocs.io/en/latest/user/quickstart/#json-response-content

This way, if the mime type is `application/json` but the response is invalid, we won't break here.

Reply:
Skipping the header check all together and just using a try/catch instead.